infra/scripts/release-patch.sh

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+
+usage()
+{
+    echo "usage: release-patch.sh
+
+    This script is used to release a patch release. It is untested on major/minor releases and for those, some modification may be necessary.
+
+    -v, --version           version to release, example: 0.10.6
+    -t, --github-token      personal GitHub token
+    -c, --starting-commit   git commit hash; the script will cherry-pick that commit and all on master after it
+"
+}
+
+while [ "$1" != "" ]; do
+  case "$1" in
+      -v | --version )           VERSION="$2";            shift;;
+      -t | --github-token )      GH_TOKEN="$2";           shift;;
+      -c | --starting-commit )   STARTING_COMMIT="$2";    shift;;
+      -h | --help )              usage;                   exit;; 
+      * )                        usage;                   exit 1
+  esac
+  shift
+done
+
+if [ -z $VERSION ]; then usage; exit 1; fi
+if [ -z $GH_TOKEN ]; then usage; exit 1; fi
+regex="([0-9]+)\.([0-9]+)\.([0-9]+)"
+if [[ $VERSION =~ $regex ]]
+then
+    MAJOR="${BASH_REMATCH[1]}"
+    MINOR="${BASH_REMATCH[2]}"
+    PATCH="${BASH_REMATCH[3]}"
+else
+    usage
+    exit 1
+fi
+
+echo "This script is mostly idempotent; check git status for temp files before restarting. It will always prompt you before making any non-local change."
+
+# Go to infra/scripts directory
+cd $(dirname "$0")
+# Login to GitHub CLI
+echo $GH_TOKEN | gh auth login --with-token
+
+echo "Step 1: Cherry-pick new commits onto release branch"
+git fetch origin
+git checkout master
+last_commit=$(git rev-parse HEAD)
+git checkout v$MAJOR.$MINOR-branch
+cherrypick_from_master()
+{
+    echo "Cherrypicking commits"
+    git cherry-pick $STARTING_COMMIT^..$last_commit
+    echo "Commits cherrypicked"
+}
+if [ -z $STARTING_COMMIT ]; then
+    read -p "No starting commit given. Skip this step (y) or exit (n)? " choice
+    case "$choice" in
+        y|Y ) echo "Skipping this step" ;;
+        n|N ) exit ;;
+        * ) exit ;;
+    esac ;
+else
+    read -p "Will cherry-pick starting from commit $STARTING_COMMIT. Continue (y) or skip this step (n)? " choice
```

Comment:
How do you cherry-pick only certain commits? Or how would the person running this script resolve issues with cherry-picks?

Reply:
Yeah my thinking here is that a common case is going to be all commits from some starting commit to the most recent one, and if that's not the case you can just cherry-pick the commits you want yourself and then run the script without specifying a starting commit in which case it'll skip the cherry-picking step. (Also feels difficult for the script to make cherry-picking arbitrary commits easier than just doing it yourself.) Issues are resolved as you normally would for a cherry-pick and then you just rerun the script and skip the cherry-picking step.